cmd/skopeo/signing.go

```diff
@@ -98,17 +104,45 @@ func (opts *standaloneVerifyOptions) run(args []string, stdout io.Writer) error
 		return fmt.Errorf("Error reading signature from %s: %v", signaturePath, err)
 	}
 
-	mech, err := signature.NewGPGSigningMechanism()
+	var mech signature.SigningMechanism
+	var fingerprints []string
+	if opts.publickeyfile != "" {
+		publicKeys, err := os.ReadFile(opts.publickeyfile)
+		if err != nil {
+			return fmt.Errorf("Error reading public keys from %s: %v", opts.publickeyfile, err)
+		}
+		mech, fingerprints, err = signature.NewEphemeralGPGSigningMechanism(publicKeys)
+	} else {
+		mech, err = signature.NewGPGSigningMechanism()
+	}
 	if err != nil {
 		return fmt.Errorf("Error initializing GPG: %v", err)
 	}
 	defer mech.Close()
+
+	if opts.publickeyfile != "" && expectedFingerprint == "any" {
+		_, expectedFingerprint, err = mech.Verify(unverifiedSignature)
+		if err != nil {
+			return fmt.Errorf("Could not determine fingerprint from signature: %v", err)
+		}
+		found := false
+		for _, fingerprint := range fingerprints {
+			if expectedFingerprint == fingerprint {
+				found = true
+			}
+		}
+		if !found {
```

Comment:
(This can be simplified to `x/exp/slices.Contains`, but it probably shouldn’t remain anyway.)

Reply:
Wasn't sure about importing the new package, but yes, it will be going.